native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractStatementParameterProcessor.java

```diff
@@ -177,6 +178,13 @@ public int setSQLValueParam(Connection connection, PreparedStatement preparedSta
             return Types.BOOLEAN;
         } else if (object instanceof BArray) {
             BArray objectArray = (BArray) object;
+
+            // If the type passed is time:Utc
+            if (objectArray.getType().toString().equals(Constants.SqlTypes.UTC)) {
```

Comment:
Let's get the name of the record and compare it.

Reply:
Seems like it is not possible. We can leave this as it is but better to verify the tuple has the required data.